audformat/core/utils.py

```diff
@@ -196,6 +196,50 @@ def concat(
         return df
 
 
+def duration(
+        obj: typing.Union[pd.Index, pd.Series, pd.DataFrame],
+) -> float:
+    r"""Duration of all segments present in a segmented object.
+
+    Args:
+        obj: object conform to
+            :ref:`table specifications <data-tables:Tables>`
+            and containing a segmented index
+
+    Returns:
+        duration in seconds
+
+    Raises:
+        ValueError: if ``obj`` does not contain a segmented index
+
+    Example:
+
+        >>> idx = segmented_index(
+        ...     files=['a', 'b', 'c'],
+        ...     starts=[0, 1, 3],
+        ...     ends=[1, 2, 4],
+        ... )
+        >>> duration(idx)
+        3.0
+
+    """
+    if index_type(obj) != define.IndexType.SEGMENTED:
+        raise ValueError('Input does not have a segmented index.')
+
+    if not isinstance(obj, pd.MultiIndex):
+        obj = obj.index
+
+    if obj.empty:
+        return 0
+
+    starts = list(obj.get_level_values(define.IndexField.START))
+    ends = list(obj.get_level_values(define.IndexField.END))
+    duration = [end - start for start, end in zip(starts, ends)]
+    duration = np.sum(duration)
+    duration = duration.total_seconds()
```

Comment:
I think we should return `duration` as a `pd.Timedelta`.

Reply:
I was afraid you would propose that ;)

Reply:
Hehehe :)

Reply:
But is there really the need here to return an object instead of a simple float value?

I guess the main argument woul dbe that `pd.Timedelta` is used at most places in `audformat` or is there also some usecase for it?

Reply:
Yes, consistency would be one argument. What I also like about `pd.Timedelta` is that the string representation is easier to read, e.g. `227348.559914` vs. `2 days 15:09:08.559914` and that you don't need have to look up in the documentation what unit the float value represents.

Reply:
Ok, I changed it to `pd.Timedelta`, but I'm pretty sure I will regret it. I'm already not able to update one of the tests as I don't know how. I will add a comment to that.